docs/tutorials/p2p.md

```diff
@@ -362,37 +321,11 @@ The key point to note here is the format **must be the same for sending and rece
 ## P2P Session Failures
 ==}
 	
-For the last part of this tutorial we'll handle P2P failures with the following function which is triggered by the `p2p_session_connect_fail` callback:
+For the last part of this tutorial we'll handle P2P failures with the following function which is triggered by the `network_messages_session_failed` callback:
 
 ```gdscript
-func _on_p2p_session_connect_fail(steam_id: int, session_error: int) -> void:
-	# If no error was given
-	if session_error == 0:
-		print("WARNING: Session failure with %s: no error given" % steam_id)
-
-	# Else if target user was not running the same game
-	elif session_error == 1:
-		print("WARNING: Session failure with %s: target user not running the same game" % steam_id)
-
-	# Else if local user doesn't own app / game
-	elif session_error == 2:
-		print("WARNING: Session failure with %s: local user doesn't own app / game" % steam_id)
-
-	# Else if target user isn't connected to Steam
-	elif session_error == 3:
-		print("WARNING: Session failure with %s: target user isn't connected to Steam" % steam_id)
-
-	# Else if connection timed out
-	elif session_error == 4:
-		print("WARNING: Session failure with %s: connection timed out" % steam_id)
-
-	# Else if unused
-	elif session_error == 5:
-		print("WARNING: Session failure with %s: unused" % steam_id)
-
-	# Else no known error
-	else:
-		print("WARNING: Session failure with %s: unknown error %s" % [steam_id, session_error])
+func _on_network_messages_session_failed(steam_id: int, session_error: int, state: int, debug_msg: String) -> void:
```

Comment:
This section should be fleshed out more to show the possible errors and what to do with the other variables.